### PR TITLE
Implement BrowserContext.OverridePermissionsAsync

### LINKED ISSUE #600: Implement BrowserContext.overridePermissions
https://github.com/kblok/puppeteer/pull/11/commits/50d6c2d3c620ad733d959f1c9b199ef932f4bc96

### PR DESCRIPTION
**Pull request by kblok**
closes #600

### CODE REVIEW

File: lib/PuppeteerSharp.Tests/PageTests/BrowserContextOverridePermissionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class BrowserContextOverridePermissionsTests : PuppeteerPageBaseTest
+    {
+        public BrowserContextOverridePermissionsTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        private Task<string> GetPermissionAsync(Page page, string name)
+            => page.EvaluateFunctionAsync<string>(
+                "name => navigator.permissions.query({ name}).then(result => result.state)",
```
Comment (Meir017) [inline lib/PuppeteerSharp.Tests/PageTests/BrowserContextOverridePermissionsTests.cs]: format 😉 
should be `name => navigator.permissions.query({ name }).then(result => result.state)`

File: lib/PuppeteerSharp/BrowserContext.cs [hunk 2]
```diff
@@ -83,6 +86,67 @@ public Task CloseAsync()
...
+        /// <example>
+        /// <![CDATA[
+        /// var context = browser.DefaultBrowserContext;
+        /// await context.OverridePermissionsAsync("https://html5demos.com", new List<string> {"geolocation"});
+        /// ]]>
+        /// </example>
+        /// <seealso href="https://developer.mozilla.org/en-US/docs/Glossary/Origin"/>
+        public async Task OverridePermissionsAsync(string origin, IEnumerable<OverridePermission> permissions)
+        {
+            var webPermissionToProtocol = new Dictionary<OverridePermission, string>
```
Comment (Meir017) [inline lib/PuppeteerSharp/BrowserContext.cs]: should this be static readonly?

File: lib/PuppeteerSharp/BrowserContext.cs [hunk 3]
```diff
@@ -83,6 +86,67 @@ public Task CloseAsync()
...
+                [OverridePermission.AccessibilityEvents] = "accessibilityEvents",
+                [OverridePermission.ClipboardRead] = "clipboardRead",
+                [OverridePermission.ClipboardWrite] = "clipboardWrite",
+                [OverridePermission.PaymentHandler] = "paymentHandler",
+                // chrome-specific permissions we have.
+                [OverridePermission.MidiSysex] = "midiSysex"
+            };
+            var permissionList = permissions.Select(permission =>
+            {
+                webPermissionToProtocol.TryGetValue(permission, out var protocolPermission);
```
Comment (Meir017) [inline lib/PuppeteerSharp/BrowserContext.cs]: maybe rewrite as
```cs
if(!webPermissionToProtocol.TryGetValue(permission, out var protocolPermission))
{
     throw ...
}
```

**Comment by Meir017** [comment]
@kblok conflicts 😢

**Comment by kblok** [comment]
Yes, closed

**Pull request closed without merging**